api/proto/teleport/legacy/types/events/events.proto

```diff
@@ -4003,6 +4003,108 @@ message DeviceEvent2 {
   ];
 }
 
+// IntegrationCreate is emitted when an integration resource is created.
+message IntegrationCreate {
+  // Metadata is a common event metadata.
+  Metadata Metadata = 1 [
+    (gogoproto.nullable) = false,
+    (gogoproto.embed) = true,
+    (gogoproto.jsontag) = ""
+  ];
+
+  // User is a common user event metadata.
+  UserMetadata User = 2 [
+    (gogoproto.nullable) = false,
+    (gogoproto.embed) = true,
+    (gogoproto.jsontag) = ""
+  ];
+
+  // ResourceMetadata is a common resource event metadata.
+  ResourceMetadata Resource = 3 [
+    (gogoproto.nullable) = false,
+    (gogoproto.embed) = true,
+    (gogoproto.jsontag) = ""
+  ];
+
+  // SubKind is the sub kind of the integration resource.
+  string SubKind = 4 [(gogoproto.jsontag) = "sub_kind"];
```

Comment:
Similar to before, let's factor this out into a separate `IntegrationMetadata` so it's easy to extend it in future if needed.

Reply:
Also @marcoandredinis are there any other integration properties we might want to include in the integration-related audit events?

Reply:
I think we should add the IAM Role and the s3 bucket info.
In case of a breach, this can be helpful to understand when the IAM Role was changed to the "bad" one (and the same goes for the bucket information).

Reply:
And I assume the same goes for the Azure OIDC integration fields (tenant and client IDs).
cc @justinas

Reply:
That's all the data in the spec for the AWS and Azure integrations. I created new protobuf messages instead of just using the subkind specs in case new spec fields are added later that we don't want added to an audit event, ie tokens or something